### PR TITLE
AP-1709 Don't send data to Geckoboard in Dev, Test, UAT environments

### PR DESCRIPTION
**Pull request by stephenrichards**
## Don't send data to Geckoboard in non-live environments

Allows devs to specify class names of jobs which should do nothing if run in development, test, staging or UAT.

- Added a Rails configuration variable `Rails.configuration.x. suspended_geckoboard_updater_jobs `, which is hash keyed by `HostEnv` environment, (`:development`, `:test`, `:staging`, `:uat`, `:production`) each containing a list of jobs which should **NOT** be run in that host environment (the values for production is an empty array, meaning all jobs are to be run.
- Added a module `Dashboard::SuspendableJob` with one method `#job_suspended?` which returns true if the class name is in the config variable for the current `HostEnv`
- Exit immediately from the job if job is suspended

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1709)

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -84,5 +84,34 @@ class Application < Rails::Application
...
+        Dashboard::ProviderDataJob
+        Dashboard::UpdaterJob
+      ],
+      uat: %w[
+        Dashboard::FeedbackItemJob
+        Dashboard::ApplicantEmailJob
+        Dashboard::ProviderDataJob
+        Dashboard::UpdaterJob
+      ],
+      staging: %w[
```
Comment (colinbruce) [inline config/application.rb:108]: I wouldn't block staging by default... 
It shouldn't be as noisy as UAT and it's often used to test functionality

**Review by colinbruce — COMMENTED** [review]
Looks good. It might be worth dropping something on the README on testing/adding new dashboard features?

File: config/application.rb [hunk 2]
```diff
@@ -84,5 +84,34 @@ class Application < Rails::Application
...
+        Dashboard::FeedbackItemJob
+        Dashboard::ApplicantEmailJob
+        Dashboard::ProviderDataJob
+        Dashboard::UpdaterJob
```
Comment (mpw5) [inline config/application.rb:112]: We might want to leave the staging jobs enabled - I know Alice sometimes tests changes there before deploying to production. With UAT it should be fairly easy to enable/disable them per environment by pushing a commit to a PR, but with staging we would have to merge and go through nearly the whole pipeline which feels time consuming/risky.

File: config/application.rb (hunk 1 above)
Comment (stephenrichards) [reply config/application.rb]: Yep, areed.  Have fixed.

File: config/application.rb (hunk 2 above)
Comment (stephenrichards) [reply config/application.rb]: Agreed

**Review by mpw5 — APPROVED** [review]
LGTM

**Pull request merged**